controllers/apps/transformer_component_load_resources.go

```diff
@@ -128,23 +130,38 @@ func (t *componentLoadResourcesTransformer) getNCheckCompDef(transCtx *component
 	return compDef, nil
 }
 
-func isGeneratedComponent(cluster *appsv1alpha1.Cluster,
-	comp *appsv1alpha1.Component) (bool, error) {
+func isGeneratedComponent(ctx context.Context, cli client.Reader, cluster *appsv1alpha1.Cluster, comp *appsv1alpha1.Component) (bool, error) {
```

Comment:
optimize or leave a TODO to remove the dependency to cluster.spec, for example, just use comp.CompDef or have the cluster pass the information through annotations

Reply:
> optimize or leave a TODO to remove the dependency to cluster.spec, for example, just use comp.CompDef or have the cluster pass the information through annotations

mark as TODO comment